lib/scryptenc/scryptenc.c

```diff
@@ -248,28 +248,32 @@ scryptenc_setup(uint8_t header[96], uint8_t dk[64],
 		/* Check logN, r, p. */
 		if ((rc = checkparams(P->maxmem, P->maxmemfrac, P->maxtime,
 		    P->logN, P->r, P->p, verbose, force)) != 0) {
+			switch(rc) {
 			/* Warn about resource limit, but suppress the error. */
```

Comment:
I think this is what you asked for, but I'm not a huge fan.

Where should the comment be, if it relates to 3 of the 4 cases?  For now, I put it on top of the first of those cases, but I think that's not as clear as the previous "comment, followed by three `if` statements".